### PR TITLE
Add hasattr function for safe if attribute is not existed.

### PR DESCRIPTION
**Pull request by geonmo**
It is just a fix to avoid missing attribute error.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SLHCUpgradeSimulations/Configuration/python/gemCustom.py [hunk 1]
```diff
@@ -9,16 +9,17 @@ def customise2023(process):
...
   from SimMuon.GEMDigitizer.customizeGEMDigi import customize_digi_addGEM_muon_only
-  process = customize_digi_addGEM_muon_only(process)
-  process.simMuonGEMDigis.mixLabel = cms.string("mix")
-  process.simMuonME0Digis.mixLabel = cms.string("mix")
-  # process.digitisation_step.remove(process.simMuonRPCDigis)
-  # process.simMuonRPCDigis.digiModel = cms.string('RPCSimParam')
-  process.simMuonRPCDigis.digiModel = cms.string('RPCSimAverageNoiseEff')
+  if ( not hasattr(process,"simMuonGEMDigis")) :
+    process = customize_digi_addGEM_muon_only(process)
+  if ( hasattr(process,"simMuonGEMDigis")) :
```
Comment (davidlange6) [inline SLHCUpgradeSimulations/Configuration/python/gemCustom.py]: this can just be an "else", right?

**Comment by geonmo** [comment]
@davidlange6 I updated about if-statement. I think that "else" is not proper at this situation.

File: SLHCUpgradeSimulations/Configuration/python/gemCustom.py [hunk 2]
```diff
@@ -9,25 +9,24 @@ def customise2023(process):
...
   from SimMuon.GEMDigitizer.customizeGEMDigi import customize_digi_addGEM_muon_only
-  process = customize_digi_addGEM_muon_only(process)
-  process.simMuonGEMDigis.mixLabel = cms.string("mix")
-  process.simMuonME0Digis.mixLabel = cms.string("mix")
-  # process.digitisation_step.remove(process.simMuonRPCDigis)
-  # process.simMuonRPCDigis.digiModel = cms.string('RPCSimParam')
-  process.simMuonRPCDigis.digiModel = cms.string('RPCSimAverageNoiseEff')
+  if ( not hasattr(process,"simMuonGEMDigis")) :
+    process = customize_digi_addGEM_muon_only(process)
+  if ( not hasattr(process,"simMuonGEMDigis.mixLabel")) :
```
Comment (davidlange6) [inline SLHCUpgradeSimulations/Configuration/python/gemCustom.py]: @geonmo - I notice only now - this is not proper python. you need to test first hasattr(process,'simMuonGEMDigis') and then hasattr(process.simMuonGEMDigis,'mixLablel')

**Comment by davidlange6** [comment]
ping @geonmo

**Comment by geonmo** [comment]
Hello, @davidlange6 
I am sorry about late reply. I fix it as your comment.
Could you test?

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
@geonmo - this does not fix at least one of my comments. Specifically

if ( not hasattr(process,"simMuonME0Digis.mixLabel")) 

will always return true. I don't think its the behavior that you want.

**Comment by geonmo** [comment]
@davidlange6 Hello! Frankly speaking, I want to avoid missing "mixLabel" error about simMuonME0Digis.mixLabel due to default setting is wrong. It will be fixed but not now. So, I added this configuration on GEM Custom.

**Comment by geonmo** [comment]
@davidlange6 If my code is correct, It will not be a problem even if simMuonME0Digis is not existed.

**Comment by davidlange6** [comment]
@geonmo - I'm not explaining well.. let me try again

hasattr(p,'x.y') is not valid python.

it will fail even if p.x.y exists.

**Comment by geonmo** [comment]
@davidlange6 Hello, I've got it your comment. I fix it properly. However, this PR is corructed any other commits from IB. I will re-submit a PR. Thanks.
